### PR TITLE
Issue/12445 budget screen redesign

### LINKED ISSUE #12445: Blaze budget screen re-design
Figma: w3JBReW1UJesCf02sj43kR-fi-2633_12836

Screenshot 2024-08-29 at 12 10 16
Screenshot 2024-08-29 at 12 10 09

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #12445

### Description

Applies the following design to Blaze budget screen: w3JBReW1UJesCf02sj43kR-fi-2633_10010

Keep in mind there are some slight design variations because the provided designs are specific to iOS. So I added some small adaptations to keep the Android style consistent.

Screenshot 2024-08-30 at 16 02 58

Screenshot 2024-08-30 at 16 03 05

### Testing information
1. Trigger Blaze campaign creation flow 
2. Click on budget section
3. Play around with the screen and check that UI is refreshed according to the designs. 
4. Special attention to switching between endless campaigns and campaigns with a specific duration. 
5. Test the UI looks good in dark mode

### The tests that have been performed

Smoke tested, updating all possible values 
Tested everything looks good in dark mode

### Images/gif

https://github.com/user-attachments/assets/95ed42b7-3319-4f6d-a919-93b8bd12486b

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -3914,26 +3914,30 @@
...
-    <string name="blaze_campaign_budget_weekly_spend">Weekly spend</string>
+    <string name="blaze_campaign_budget_duration_subtitle">How much would you like to spend on your campaign, and how long should it run for?</string>
+    <string name="blaze_campaign_budget_footer_weekly_spend">weekly spend</string>
+    <string name="blaze_campaign_budget_daily_spend_label">Daily spend</string>
     <string name="blaze_campaign_budget_days_duration">for %1$s days</string>
     <string name="blaze_campaign_budget_daily_spend">%1$s daily</string>
     <string name="blaze_campaign_budget_reach_forecast">Estimated people reached per day</string>
-    <string name="blaze_campaign_budget_duration_section_title">Duration</string>
-    <string name="blaze_campaign_budget_duration_section_endless_campaign_value">Starting from %1$s</string>
+    <string name="blaze_campaign_budget_scheduled_section_title">Scheduled</string>
```
Comment (irfano) [inline WooCommerce/src/main/res/values/strings.xml]: The design uses "Schedule." Did you use "Scheduled" intentionally?

**Review by irfano — CHANGES_REQUESTED** [review]
**Suggestion:** I noticed an issue with the date picker that's beyond the scope of this PR. When the date picker is opened, it always defaults to today's date instead of the currently selected Starts date. I believe it should display the selected Starts date. Would you like to fix this in this PR, or want me to open an issue for it?

**Bug:** Another issue is that if I change the state of the bottom sheet and then dismiss it, the state is retained when it shouldn’t be.
<details>
  <summary>bug video for bottom sheet state</summary>
 
[bug.webm](https://github.com/user-attachments/assets/7b6cdcf6-10e4-4225-a85a-9bd7b6d72b71)
</details>

**Suggestion:** The design uses the thousand separators but we don't use it in the app. I suggest adding it.

image

**Comment by JorgeMucientes** [comment]
Excellent feedback and attention to details @irfano. Thanks. I've addressed all of it. Ready for another round.

**Review by irfano — CHANGES_REQUESTED** [review]
Thanks for addressing my comments. 

- When I change the duration switch and dismiss the bottom sheet, the animation stucks halfway.  Could you please take a look at this issue?
<details>
  <summary>Bottom sheet animation issue</summary>
 
[Screen_recording_20240903_150317.webm](https://github.com/user-attachments/assets/07da8c46-9c30-497e-9725-8a65fb63e363)
</details>

- The bottom sheet still retains the slider’s value.
<details>
  <summary>Bottom sheet slider value issue</summary>
 
[bug2.webm](https://github.com/user-attachments/assets/bba9cd9a-103a-4118-abd3-0bfca1c2839b)
</details>

**Comment by JorgeMucientes** [comment]
Thanks for the feedback @irfano 

> When I change the duration switch and dismiss the bottom sheet, the animation stucks halfway. Could you please take a look at this issue?

Yeah, I was aware of this issue when submitting the PR. While I spent some time trying to mitigate it, I failed and though speding to much time on it, wasn't worth it. After your feedback I gave it some more thinking. I added a workaround to ensure animations run one after the other, and the weird effect is gone. I still feel it is a bit weird that the user will see how all the values they changed are reverted to the original ones before the bottom sheet is hidden. But I think that is an acceptable behavior. 

Let me know your thoughts. 

> The bottom sheet still retains the slider’s value.

Arg completely missed that one, sorry. Fixed it.

**Review by irfano — APPROVED** [review]
It's better now. 👍🏻 If you told me it wasn’t a bug but a feature, I would believe it. 😆 Users can now see that their choices are not saved.

[bug.webm](https://github.com/user-attachments/assets/e2289969-9ee5-4029-a530-a1569c708c32)

LGTM!

**Pull request merged**